src/python/pants/backend/native/tasks/native_compile.py

```diff
@@ -209,7 +218,7 @@ def _compile(self, compile_request):
     """
     sources = compile_request.sources
 
-    if len(sources) == 0:
+    if len(sources) == 0 or not [s for s in sources if not s.endswith(('.h', '.hpp'))]:
```

Comment:
It would be good to lift these into a constant (or use an existing one, perhaps).

But also, rather than using `not [..]` (ie "list is empty"), prefer to use `any(.. for .. in ..)`.

Reply:
✅